src/NServiceBus.Persistence.AzureTable/Outbox/OutboxRecord.cs

```diff
@@ -29,7 +29,14 @@ public string Id
 
         public override IDictionary<string, EntityProperty> WriteEntity(OperationContext operationContext)
         {
-            TransportOperations = JsonConvert.SerializeObject(Operations, Formatting.Indented);
+            TransportOperations = JsonConvert.SerializeObject(
```

Comment:
Might want to try using a custom converter like one [here](https://github.com/Particular/NServiceBus.Persistence.Sql/pull/774/files#diff-a4102449d3b5a9301cb56da745b64f6ba8b34af33fb8d6c34509eaf492607456) to avoid the `ToArray`. But actually I am not sure how `ToArray` works here i.e. we have created the `ReadOnlyMemory<byte>` from an array so maybe `ToArray` just returns the underlying buffer? 

I used the custom converter in SQLP and Cosmos because I figured it might save the re-allocation of the data but I did not bother to use it in RavenDB nor Mongo.

Reply:
ToArray allocates new array. Instead of converter, I used MemoryMarshal.TryGetArray which returns array representing the memory (without allocation), I am not sure when this method can fail, but when it does, I am doing ToArray()